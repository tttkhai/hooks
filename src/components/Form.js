import React from "react";

function Form({ text, handleChange, handleSubmit }) {
  const change = (value) => {
    handleChange(value);
  };

  const submit=(e)=>{
    e.preventDefault();
    handleSubmit();
  }

  return (
    <div>
      <form
        onSubmit={(e) => submit(e)}
      >
        <input
          value={text}
          name="text"
          onChange={(e) => change(e.target.value)}
        />
      </form>
    </div>
  );
}

export default Form;
