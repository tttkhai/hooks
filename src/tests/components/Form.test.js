import React from "react";
import { shallow } from "enzyme";
import Form from "../../components/Form";
  

test('test form submission', ()=>{
  const wrapper = shallow(<Form />);
  expect(wrapper).toMatchSnapshot()
  wrapper.find('form').simulate('submit', {
    preventDefault: ()=> {}
  })

  expect(wrapper.state('error').length).toBeGreaterThan(0)
  expect(wrapper).toMatchSnapshot()
});

test('test form change input', ()=>{
  const value="123"
  const wrapper = shallow(<Form />);
  expect(wrapper).toMatchSnapshot()
  wrapper.find('input').at(0).simulate('change', {
    target: {value}
  })

  expect(wrapper.state('text')).toBe(value)
  expect(wrapper).toMatchSnapshot()
});


