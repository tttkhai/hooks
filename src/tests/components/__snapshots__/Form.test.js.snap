// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test form change input 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="text"
      onChange={[Function]}
    />
  </form>
</div>
`;

exports[`test form submission 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="text"
      onChange={[Function]}
    />
  </form>
</div>
`;

exports[`test form submission 2`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      name="text"
      onChange={[Function]}
    />
  </form>
</div>
`;
